### PR TITLE
persist: close Buffers and Blobs during shutdown so they release LOCKs

### PR DESCRIPTION
**Pull request by danhhz**

### CODE REVIEW

File: src/persist/src/file.rs [hunk 1]
```diff
@@ -115,9 +116,11 @@ impl FileBuffer {
 
             if len > 0 {
                 let mut buf = [0u8; 8];
-                buffer_file.seek(SeekFrom::End(8))?;
+                buffer_file.seek(SeekFrom::End(-8))?;
```
Comment (ruchirK) [inline src/persist/src/file.rs:119]: 🤦🏾 whoops sorry

File: src/persist/src/file.rs [hunk 2]
```diff
@@ -152,10 +156,18 @@ impl FileBuffer {
...
+    fn ensure_open(&self) -> Result<PathBuf, Error> {
+        self.base_dir
+            .clone()
+            .ok_or_else(|| return Error::from("FileBuffer unexpectedly closed"))
+    }
 }
 
 impl Buffer for FileBuffer {
     fn write_sync(&mut self, buf: Vec<u8>) -> Result<SeqNo, Error> {
+        let _ = self.ensure_open()?;
```
Comment (ruchirK) [inline src/persist/src/file.rs:169]: I think this can just be `self.ensure_open()?;`

**Review by ruchirK — APPROVED** [review]
LGTM!

File: src/persist/src/file.rs (hunk 1 above)
Comment (danhhz) [reply src/persist/src/file.rs:119]: Don't be, it passed the buffer tests! Now there are just more tests

File: src/persist/src/file.rs (hunk 2 above)
Comment (danhhz) [reply src/persist/src/file.rs:169]: Done

**Review by danhhz — COMMENTED** [review]
TFTR! Merging on green

**Pull request merged**